streams/src/main/java/org/apache/kafka/streams/state/TimestampedSessionStore.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state;
+
+import org.apache.kafka.streams.kstream.Windowed;
+
+/**
+ * Interface for storing the aggregated values of sessions.
+ *
+ * In contrast to a {@link SessionStore} that stores windowedKeys-value pairs,
+ * a {@code TimestampedSessionStore} stores windowedKeys-(value/timestamp) pairs.
```

Comment:
We should probably emphasize the difference between the timestamps used in windowed key and the timestamps associated with the value here, as they will be very confusing to users. Ditto for window store below.

Reply:
Ack.

Note, I also updated the JavaDocs of `WindowStore` and `SessionStore`.